src/platform/unix/mod.rs

```diff
@@ -1054,10 +1060,23 @@ impl UnixCmsg {
 
     unsafe fn recv(&mut self, fd: c_int, blocking_mode: BlockingMode)
                    -> Result<usize, UnixError> {
-        if let BlockingMode::Nonblocking = blocking_mode {
-            if libc::fcntl(fd, libc::F_SETFL, libc::O_NONBLOCK) < 0 {
-                return Err(UnixError::last())
+        match blocking_mode {
+            BlockingMode::Nonblocking => {
+                if libc::fcntl(fd, libc::F_SETFL, libc::O_NONBLOCK) < 0 {
+                    return Err(UnixError::last())
+                }
+            },
+            BlockingMode::Timeout(duration) => {
+                let events = libc::POLLIN | libc::POLLPRI | libc::POLLRDHUP;
+                let fd = &mut [libc::pollfd {fd, events, revents: 0}];
```

Comment:
Let's write this as `let mut fd = [...]` instead. I find that easier to reason about.

Reply:
Resolved in https://github.com/servo/ipc-channel/pull/284/commits/da99fcae7d890761881f8e0bbf035c4f995d357e